src/darwin/Framework/CHIP/MTRDemuxingStorage.h

```diff
@@ -0,0 +1,109 @@
+/**
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+#import <Matter/MTRDeviceControllerFactory.h>
+
+#include <lib/core/CHIPPersistentStorageDelegate.h>
+#include <lib/core/DataModelTypes.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ * A PersistentStorageDelegate implementation that does the following:
+ *
+ * 1) Ensures that any "global" storage keys are stored in RAM as needed so that
+ *    the Matter stack has access to them.
+ * 2) Hands off fabric-index-specific keys to the controller that corresponds to
+ *    that fabric index, if any.
+ */
+class MTRDemuxingStorage : public chip::PersistentStorageDelegate {
+public:
+    MTRDemuxingStorage(MTRDeviceControllerFactory * factory);
+    ~MTRDemuxingStorage() {}
+
+    // PersistentStorageDelegate API.
+    CHIP_ERROR SyncGetKeyValue(const char * key, void * buffer, uint16_t & size) override;
+
+    CHIP_ERROR SyncSetKeyValue(const char * key, const void * value, uint16_t size) override;
+
+    CHIP_ERROR SyncDeleteKeyValue(const char * key) override;
+
+private:
+    static bool IsGlobalKey(NSString * key);
```

Comment:
Any reason these need to be class methods if they are only consumed in the implementation?

Reply:
No particular reason, probably.  I started out with them being instance methods before it turned out they did not need any instance state.  I can move them to just be static methods in the impl file.